### PR TITLE
Fix MyPy Errors for provider Snowflake

### PR DESCRIPTION
**Pull request by kazanzhy**
Fixed a few MyPy errors for the 
* Snowflake provider

### CODE REVIEW

File: airflow/providers/snowflake/hooks/snowflake.py [hunk 1]
```diff
@@ -135,7 +137,7 @@ def __init__(self, *args, **kwargs) -> None:
         self.schema = kwargs.pop("schema", None)
         self.authenticator = kwargs.pop("authenticator", None)
         self.session_parameters = kwargs.pop("session_parameters", None)
-        self.query_ids = []
+        self.query_ids: List[str] = []
```
Comment (josh-fell) [inline airflow/providers/snowflake/hooks/snowflake.py:138]: Is the `self.query_ids = []` needed in the `run()` method if it's already defined here?

File: airflow/providers/snowflake/hooks/snowflake.py [hunk 2]
```diff
@@ -64,6 +64,8 @@ class SnowflakeHook(DbApiHook):
     :param session_parameters: You can set session-level parameters at
         the time you connect to Snowflake
     :type session_parameters: Optional[dict]
+    :param query_ids:
```
Comment (josh-fell) [inline airflow/providers/snowflake/hooks/snowflake.py]: There should be a description here. The API docs for this hook are built from the docstring and we shouldn't have a parameter without a description.

File: airflow/providers/snowflake/hooks/snowflake.py [hunk 3]
```diff
@@ -239,6 +247,8 @@ def run(self, sql: Union[str, list], autocommit: bool = False, parameters: Optio
         :type autocommit: bool
         :param parameters: The parameters to render the SQL query with.
         :type parameters: dict or iterable
+        :param handler: The result handler which is called with the result of each statement.
+        :type handler: dict or iterable
```
Comment (josh-fell) [inline airflow/providers/snowflake/hooks/snowflake.py]: The docstring doesn't match the typing here.

File: airflow/providers/snowflake/hooks/snowflake.py (hunk 2 above)
Comment (uranusjr) [reply airflow/providers/snowflake/hooks/snowflake.py]: I think this is simply wrong? The hook has an attribute named `query_ids`, but does not accept `query_ids` as an argument.

**Review by uranusjr — DISMISSED** [review]

File: airflow/providers/snowflake/hooks/snowflake.py (hunk 1 above)
Comment (kazanzhy) [reply airflow/providers/snowflake/hooks/snowflake.py:138]: I assume that in `run()` method this statement resets the list with ids.
So it keeps just last run ids. I think that's reasonable

File: airflow/providers/snowflake/hooks/snowflake.py (hunk 3 above)
Comment (kazanzhy) [reply airflow/providers/snowflake/hooks/snowflake.py]: Fixed

File: airflow/providers/snowflake/hooks/snowflake.py (hunk 2 above)
Comment (josh-fell) [reply airflow/providers/snowflake/hooks/snowflake.py]: Oh duh, you’re right, that was silly of me to even suggest.

File: airflow/providers/snowflake/hooks/snowflake.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/snowflake/hooks/snowflake.py:138]: Agree. this makes the hook re-runnable/

File: airflow/providers/snowflake/operators/snowflake.py [hunk 4]
```diff
@@ -230,7 +230,7 @@ def __init__(
         self.schema = schema
         self.authenticator = authenticator
         self.session_parameters = session_parameters
-        self.query_ids = []
+        self.query_ids: List[str] = []
```
Comment (potiuk) [inline airflow/providers/snowflake/operators/snowflake.py:233]: Strange that it's not inferred automatically here. But it does not hurt.

**Review by potiuk — APPROVED** [review]

File: airflow/providers/snowflake/operators/snowflake.py (hunk 4 above)
Comment (potiuk) [reply airflow/providers/snowflake/operators/snowflake.py:233]: But maybe you can remove it since the docstring needs to be fixed.

**Comment by potiuk** [comment]
> I think this is simply wrong? The hook has an attribute named query_ids, but does not accept query_ids as an argument.
Yep. it is wrong. Should be removed.

**Review by potiuk — CHANGES_REQUESTED** [review]
The docstring + possibly redundant type specification should be removed

File: airflow/providers/snowflake/operators/snowflake.py (hunk 4 above)
Comment (kazanzhy) [reply airflow/providers/snowflake/operators/snowflake.py:233]: Yeah, unfortunately, it doesn't infer automatically. 
I tried without this hint but mypy doesn't like it.

**Review by potiuk — APPROVED** [review]

**Comment by kazanzhy** [comment]
Hi. I'm sorry for this force-push, but I saw that I'm applying `handler` for every row. 
But in according to DbApiHook it should be applied to the cursor.
```
if handler is not None:
    cur = handler(cur)
```

**Review by potiuk — APPROVED** [review]

**Pull request merged**